flask_login/login_manager.py

```diff
@@ -378,26 +378,20 @@ def _load_from_cookie(self, cookie):
             user_loaded_from_cookie.send(app, user=_get_user())
 
     def _load_from_header(self, header):
-        user = None
-        if self.header_callback:
-            user = self.header_callback(header)
+        user = self.header_callback(header)
+        self.reload_user(user=user)
         if user is not None:
-            self.reload_user(user=user)
+            session['user_id'] = getattr(user, self.id_attribute)()
```

Comment:
Can you add a new test that fails when `user_id` is not set on `session` and passes when it is set? The test should explain why the `user_id` needs to be set on `session` because everything seems to be working fine right now according to tests.